### ISSUE TITLE
Disable History and Rollback and Delete Options for certain users 

### ISSUE THREAD

**Issue by rajivml** [argoproj/argo-cd#6916]
Hi,

Is there a way we can disable, "History and Rollback" and "Delete" options.

![image](https://user-images.githubusercontent.com/46923567/128330279-796e9d23-92e7-4c71-919a-d5051f9eb09b.png)

Motivation:
We want to disable "Rollback" because we don't have the rollback logic coded into our deployments so if any of our end users try to use this, the app might go into inconsistent state and we want to avoid that

I had a look at the RBAC policy and I don't find a way to disable the same https://github.com/argoproj/argo-cd/blob/master/assets/builtin-policy.csv

**Comment by dmitry-mightydevops** [comment]
I don't think that is possible at this time. You can only grant access to applications within the project with "ALL" functionality allowed
See more: https://github.com/argoproj/argo-cd/issues/6916

**Comment by raolivei** [comment]
> I don't think that is possible at this time. You can only grant access to applications within the project with "ALL" functionality allowed See more: #6916

#6916 is this very same issue.

**Comment by raolivei** [comment]
@dmitry-mightydevops   Is it possible to disable History and Rollback to **all** users?

**Comment by taliastocks** [comment]
Watching with the hope of more granular RBAC permissions in the future.

**Comment by crenshaw-dev** [comment]
You can disable rollback access by disabling `update` access.

If you'd like to disable rollback but leave update access, this issue covers that request: https://github.com/argoproj/argo-cd/issues/9936
